server/Arcadia.Assistant.Calendar/PendingActions/EmployeeVacationsPendingActionsActor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Arcadia.Assistant.Calendar.PendingActions
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Akka.Actor;
+    using Akka.Event;
+
+    using Arcadia.Assistant.Calendar.Abstractions;
+    using Arcadia.Assistant.Calendar.Abstractions.Messages;
+
+    public class EmployeeVacationsPendingActionsActor : UntypedActor, ILogReceive
+    {
+        private readonly string employeeId;
+        private readonly List<CalendarEvent> pendingActionEvents = new List<CalendarEvent>();
+
+        public EmployeeVacationsPendingActionsActor(string employeeId)
+        {
+            this.employeeId = employeeId;
+
+            Context.System.EventStream.Subscribe<CalendarEventApproverEventBusMessage>(this.Self);
+        }
+
+        public static Props CreateProps(string employeeId)
+        {
+            return Props.Create(() => new EmployeeVacationsPendingActionsActor(employeeId));
+        }
+
+        protected override void OnReceive(object message)
+        {
+            switch (message)
+            {
+                case GetEmployeePendingActions _:
+                    this.Sender.Tell(new GetEmployeePendingActions.Response(this.pendingActionEvents));
+                    break;
+
+                case CalendarEventApproverEventBusMessage msg:
+                    if (msg.Event.Type != CalendarEventTypes.Vacation)
+                    {
+                        break;
+                    }
+
+                    if (msg.ApproverId == this.employeeId &&
+                        this.pendingActionEvents.All(e => e.EventId != msg.Event.EventId))
```

Comment:
Dictionary instead?

Reply:
Actor response shouldn't depend on what caller wants to do with it. For now we show it to user as dictionary, later we'll want to show all events in the order they were added.